src/planner/expression_binder/order_binder.cpp

```diff
@@ -100,6 +101,24 @@ unique_ptr<Expression> OrderBinder::Bind(unique_ptr<ParsedExpression> expr) {
 	case ExpressionClass::PARAMETER: {
 		throw ParameterNotAllowedException("Parameter not supported in ORDER BY clause");
 	}
+	case ExpressionClass::COLLATE: {
+		auto &collation = expr->Cast<CollateExpression>();
+		if (collation.child->expression_class == ExpressionClass::CONSTANT) {
+			auto &constant = collation.child->Cast<ConstantExpression>();
+			auto index = (idx_t)constant.value.GetValue<int64_t>() - 1;
```

Comment:
I believe this needs a `NumericCast`, and we should add a test with a negative value